<!-- 

-->

<launch>

<arg name="bagfile_path"  value="$(env ROS_DATASET_PATH)/12Jul2015SSRR/1.bag" />

<param name="use_sim_time" value="false" />

<node pkg ="rosbag"
    name  ="rosbag"
    type  ="play"
    output="screen"
    args  ="--loop --rate=0.55 $(arg bagfile_path) --start=30 --duration=9999 --delay=5 ">
    <!-- <remap from="/cam2/image" to="/dummy1" /> -->
</node>

<node pkg = "image_transport"
	  name = "image_transport"
	  type = "republish"
      args = "compressed in:=/cam2/image raw out:=/camera/image_raw"
	  output = "screen" >
</node>

<node pkg   ="image_view"
	  name  ="image_view_raw"
	  type  ="image_view"
	  output="screen">

	  <remap from="/image" to="/camera/image_raw"/>
</node>

<node pkg   ="image_view"
	  name  ="image_view_of_field"
	  type  ="image_view"
	  output="screen">

	  <remap from="/image" to="/of_estimator_node/of_image"/>
</node>

<node pkg   ="image_proc"
	  name  ="image_proc"
	  type  ="image_proc"
	  ns    ="/camera"
	  output="screen">
</node>

<node pkg   ="of_estimator"
      name  ="of_estimator_node"
      type  ="of_estimator_node"
      output="screen">
                         
  	  <param name="debug_mode"    value="false" />
	  <param name="grid_cols"     value="10"   />
	  <param name="grid_rows"     value="8"    />
	  <param name="pyra_levels"   value="3"     />
	  <param name="use_init_flow" value="false" />
	  <remap from="~image_raw" to="/camera/image_raw"/>
</node>


</launch>



